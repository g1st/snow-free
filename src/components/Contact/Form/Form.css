.Form {
  margin: 30px 10px 80px 10px;
}

.Row {
  display: flex;
  flex-flow: column;
}

input {
  box-sizing: border-box;
  display: block;
  width: 100%;
  padding: 15px 20px;
  margin-bottom: 30px;
  border: none;
  letter-spacing: 1px;
  outline: none;
}

textarea {
  letter-spacing: 1px;
  outline: none;
  border: none;
  box-sizing: border-box;
  width: 100%;
  padding: 15px 20px;
  margin-bottom: 30px;
}

@media (min-width: 760px) {
  .Name {
    margin-right: 15px;
  }
  .Email {
    margin-left: 15px;
  }
  .Row {
    flex-flow: row;
  }
}