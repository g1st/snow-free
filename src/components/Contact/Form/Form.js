import React, { Component } from 'react';
import Button from 'material-ui/Button';
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles';

import classes from './Form.css';

const theme = createMuiTheme({
  palette: {
    primary: { main: '#252525' },
  },
});

class Form extends Component {
  state = {
    textarea: '',
    name: '',
    email: '',
    title: '',
  };

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit(event) {
    event.preventDefault();
    alert(`Comment was sent: ${this.state.textarea}`);
  }

  render() {
    return (
      <div className={classes.Form}>
        <form onSubmit={e => this.handleSubmit(e)}>
          <div className={classes.Row}>
            <input
              className={classes.Name}
              required
              type="text"
              name="name"
              placeholder="Your Name"
              value={this.state.name}
              onChange={e => this.handleChange(e)}
            />
            <input
              className={classes.Email}
              required
              type="email"
              name="email"
              placeholder="Your Email"
              value={this.state.email}
              onChange={e => this.handleChange(e)}
            />
          </div>
          <input
            type="text"
            name="title"
            placeholder="Your Title"
            value={this.state.title}
            onChange={e => this.handleChange(e)}
          />
          <textarea
            name="textarea"
            value={this.state.textarea}
            placeholder="Your Comment"
            rows="8"
            onChange={e => this.handleChange(e)}
            required
          />
          <div className={classes.Button}>
            <MuiThemeProvider theme={theme}>
              <Button
                variant="raised"
                color="primary"
                size="large"
                type="submit"
              >
                Send message
              </Button>
            </MuiThemeProvider>
          </div>
        </form>
      </div>
    );
  }
}

export default Form;
