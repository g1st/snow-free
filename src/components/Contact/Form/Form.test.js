import React from 'react';
import renderer from 'react-test-renderer';

import Form from './Form.js';

describe('Form component renders the form correctly', () => {
  it('renders correctly', () => {
    const rendered = renderer.create(<Form />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
