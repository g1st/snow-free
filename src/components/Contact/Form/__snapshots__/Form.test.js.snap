// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders the form correctly renders correctly 1`] = `
<div
  className="Form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="Row"
    >
      <input
        className="Name"
        name="name"
        onChange={[Function]}
        placeholder="Your Name"
        required={true}
        type="text"
        value=""
      />
      <input
        className="Email"
        name="email"
        onChange={[Function]}
        placeholder="Your Email"
        required={true}
        type="email"
        value=""
      />
    </div>
    <input
      name="title"
      onChange={[Function]}
      placeholder="Your Title"
      type="text"
      value=""
    />
    <textarea
      name="textarea"
      onChange={[Function]}
      placeholder="Your Comment"
      required={true}
      rows="8"
      value=""
    />
    <div
      className="Button"
    >
      <button
        className="MuiButtonBase-root-16 MuiButton-root-1 MuiButton-raised-6 MuiButton-raisedPrimary-7 MuiButton-sizeLarge-14"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-2"
        >
          Send message
        </span>
        <span
          className="MuiTouchRipple-root-19"
        />
      </button>
    </div>
  </form>
</div>
`;
